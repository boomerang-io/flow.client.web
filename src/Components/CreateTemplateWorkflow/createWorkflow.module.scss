@import "Styles/mixins";

.disabledCreate {
  align-items: center;
  border: 0.0625rem solid #a2a9b0;
  box-sizing: border-box;
  @include border-radius();
  display: flex;
  font-family: "IBM Plex Sans";
  height: 10.325rem;
  justify-content: center;
  line-height: 1;
  padding: 0.125rem;
  position: relative;
  width: 19.375rem;
  z-index: 1;

  svg {
    fill: #a2a9b0;
  }
  p {
    color: #a2a9b0;
  }
}

.container {
  align-items: center;
  border: 0.0625rem solid #373d42;
  box-sizing: border-box;
  @include border-radius();
  cursor: pointer;
  display: flex;
  font-family: "IBM Plex Sans";
  height: 10.325rem;
  justify-content: center;
  line-height: 1;
  padding: 0.125rem;
  position: relative;
  width: 19.375rem;
  z-index: 1;

  &:focus {
    @include focus-outline();
  }
}

.modalContainer {
  height: 46rem;
  max-height: 84vh;
  min-width: 55rem;
}

.addIcon {
  height: 1rem;
  width: 1rem;
}

.text {
  font-size: 0.75rem;
  margin-left: 0.5rem;
}
