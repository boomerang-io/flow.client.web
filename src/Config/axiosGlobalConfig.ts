// Setup global config for Axios requests so they work with ICP
import axios from "axios";
axios.defaults.withCredentials = true;
