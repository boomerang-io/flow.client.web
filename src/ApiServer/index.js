import { Server, Serializer, Model, Response } from "miragejs";
import { inflections } from "inflected";
import queryString from "query-string";
import { v4 as uuid } from "uuid";
import { serviceUrl, BASE_URL } from "Config/servicesConfig";
import * as fixtures from "./fixtures";

export function startApiServer({ environment = "test", timing = 0 } = {}) {
  inflections("en", function (inflect) {
    // Prevent pluralization bc our apis are weird
    inflect.irregular("activity", "activity");
    inflect.irregular("config", "config");
    // inflect.irregular("tasktemplates", "tasktemplates");
    inflect.irregular("insights", "insights");
    inflect.irregular("flowNavigation", "flowNavigation");
  });

  return new Server({
    environment,
    // Load in mock data
    fixtures,
    // Return the data as is, don't add a root key
    serializers: {
      application: Serializer.extend({
        root: false,
        embed: true,
      }),
    },
    // Register the data as a model so we can use the schema
    models: {
      activity: Model,
      approverGroups: Model,
      changelog: Model,
      globalParams: Model,
      featureFlag: Model,
      insights: Model,
      manageTeam: Model,
      manageTeamDetail: Model,
      manageUser: Model,
      quotas: Model,
      revision: Model,
      setting: Model,
      summary: Model,
      taskTemplates: Model,
      team: Model,
      teamApproverUsers: Model,
      teamNameValidate: Model,
      taskTemplateValidate: Model,
      teamProperties: Model,
      tokens: Model,
      flowNavigation: Model,
      workflowCalendar: Model,
      workflowSchedules: Model,
    },

    routes() {
      // Control how long the responses take to resolve
      this.timing = timing;

      // Allow unhandled requests on the current domain to pass through
      this.passthrough();

      this.get("/info", () => []);

      /**
       * Simple GET of static data
       */
      this.get(serviceUrl.getUserProfile(), (schema) => {
        return schema.db.profile[0];
      });

      this.get(serviceUrl.getContext(), (schema) => {
        return schema.db.platformConfig[0];
      });

      this.get(`${BASE_URL}/navigation`, (schema) => {
        return schema.db.flowNavigation;
      });

      this.get(serviceUrl.getMyTeams({ query: null }), (schema) => {
        return schema.db.myTeams[0];
      });

      this.get(serviceUrl.getFeatureFlags(), (schema) => {
        return schema.db.featureFlags[0];
      });

      this.get(
        serviceUrl.getWorkflowSchedulesCalendar({
          workflowId: ":workflowId",
          query: null,
        }),
        (schema) => {
          return schema.db.workflowCalendar;
        }
      );

      this.get(serviceUrl.getSchedule({ query: null }), (schema) => {
        return schema.db.workflowSchedules;
      });

      this.get(serviceUrl.getSchedules({ query: null }), (schema) => {
        return schema.db.workflowSchedules[0];
      });

      this.get(serviceUrl.getWorkflowSchedules({ workflowId: ":workflowId" }), (schema) => {
        return schema.db.workflowSchedules;
      });

      this.get(serviceUrl.workflowAvailableParameters({ workflowId: ":workflowId" }), (schema) => {
        return schema.db.availableParameters[0].data;
      });

      this.get(serviceUrl.getWorkflowTemplates(), (schema) => {
        return schema.db.workflowTemplates[0];
      });

      this.get(serviceUrl.getWorkflows({ query: null }), (schema) => {
        return schema.db.workflows[0];
      });

      this.get(serviceUrl.getTeamQuotas({ id: ":id" }), (schema) => {
        return schema.db.quotas[0];
      });

      this.get(serviceUrl.getTeams({ query: null }), (schema) => {
        return schema.db.teams[0];
      });

      this.put(serviceUrl.putActivationApp(), () => {
        return {};
      });

      /**
       * Global Parameters
       */

      this.get(serviceUrl.getGlobalParams({ query: null }), (schema) => {
        return schema.db.globlaParams[0];
      });
      this.post(serviceUrl.getGlobalParams(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        schema.globalParams.create({ id: uuid(), ...body });
        return schema.globalParams.all();
      });

      this.patch(serviceUrl.getGlobalProperty({ id: ":id" }), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let { id } = request.params;
        let param = schema.globalParams.find(id);
        param.update({ ...body });
      });

      this.delete(serviceUrl.getGlobalProperty({ id: ":id" }), (schema, request) => {
        let { id } = request.params;
        schema.db.globalParams.remove({ id });
      });

      /**
       * Team Propertiies
       */
      this.get(serviceUrl.getTeamProperties({ id: ":id" }), (schema, request) => {
        let { id } = request.params;
        let property = schema.teamProperties.find(id);
        return property && property.properties ? property.properties : [];
      });
      this.post(serviceUrl.getTeamProperties({ id: ":id" }), (schema, request) => {
        /**
         * find team record, update the list of properties for that team
         */
        let { id } = request.params;
        let body = JSON.parse(request.requestBody);
        let activeTeamProperty = schema.teamProperties.find(id);
        let currentProperties = activeTeamProperty.attrs.properties;
        currentProperties.push({ id: uuid(), ...body });
        activeTeamProperty.update({ properties: currentProperties });
        return schema.teamProperties.all();
      });
      this.patch(
        serviceUrl.getTeamProperty({ teamId: ":teamId", configurationId: ":configurationId" }),
        (schema, request) => {
          /**
           * find team record, update the list of properties for that team
           */
          let { teamId, configurationId } = request.params;
          let body = JSON.parse(request.requestBody);
          let activeTeamProperty = schema.teamProperties.find(teamId);
          let currentProperties = activeTeamProperty.attrs.properties;
          let foundIndex = currentProperties.findIndex((prop) => prop.id === configurationId);
          currentProperties[foundIndex] = body;
          activeTeamProperty.update({ properties: currentProperties });
          return schema.teamProperties.all();
        }
      );
      this.delete(
        serviceUrl.getTeamProperty({ teamId: ":teamId", configurationId: ":configurationId" }),
        (schema, request) => {
          /**
           * find team record, update the list of properties for that team
           */
          let { teamId, configurationId } = request.params;
          let activeTeamProperty = schema.teamProperties.find(teamId);
          let currentProperties = activeTeamProperty.attrs.properties;
          let newProperties = currentProperties.filter((prop) => prop.id !== configurationId);
          activeTeamProperty.update({ properties: newProperties });
          return schema.teamProperties.all();
        }
      );

      /**
       * Insights
       */
      this.get(serviceUrl.getInsights({ query: null }), (schema, request) => {
        //grab the querystring from the end of the request url
        const query = request.url.substring(14);
        // eslint-disable-next-line
        const { fromDate = null, toDate = null, teamId = null } = queryString.parse(query);
        const activeTeam = teamId && schema.db.myTeams.find(teamId);
        let activeExecutions =
          activeTeam && schema.db.insights[0].executions.filter((team) => team.teamName === activeTeam.name);
        return activeExecutions ? { ...schema.db.insights[0], executions: activeExecutions } : schema.db.insights[0];
      });

      /**
       * Task Templates
       */
      const tasktemplatePath = serviceUrl.getTaskTemplates({ query: null });
      this.get(serviceUrl.getTaskTemplates({ query: null }), (schema) => {
        console.log(schema.db.taskTemplates);
        return schema.db.taskTemplates[0];
      });
      this.put(tasktemplatePath, (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let taskTemplate = schema.tasktemplate.find(body.id);
        taskTemplate.revisions.push(body);
        taskTemplate.update({ ...body });
        return taskTemplate;
      });

      this.post(serviceUrl.postValidateYaml(), (schema) => {
        return schema.db.taskTemplateValidate[0];
      });

      /**
       * Workflows
       */

      // Workflow Summary
      this.get(serviceUrl.getWorkflowSummary({ workflowId: ":workflowId" }), (schema, request) => {
        let { workflowId } = request.params;
        return schema.summaries.find(workflowId);
      });

      this.patch(serviceUrl.patchUpdateWorkflowSummary(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let summary = schema.summaries.find(body.workflowId);
        summary.update(body);
        return summary;
      });

      this.post(serviceUrl.postCreateWorkflow(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let workflow = { ...body, id: uuid(), createdDate: Date.now(), revisionCount: 1, status: "active" };
        if (body.flowTeamId) {
          let flowTeam = schema.myTeams.findBy({ id: body.flowTeamId });
          const teamWorkflows = [...flowTeam.workflows];
          teamWorkflows.push(workflow);
          flowTeam.update({ workflows: teamWorkflows });
          return schema.summaries.create(workflow);
        }
        return {};
      });

      this.post(
        serviceUrl.postCreateWorkflowToken({ workflowId: ":workflowId", label: ":label" }),
        (schema, request) => {
          return { token: uuid() };
        }
      );

      this.del(serviceUrl.getWorkflow({ id: ":workflowId" }), (schema, request) => {
        let { workflowId } = request.params;
        let flowTeam = schema.myTeams.where((team) => team.workflows.find((workflow) => workflow.id === workflowId));
        let { attrs } = flowTeam.models[0];
        const teamWorkflows = attrs.workflows.filter((workflow) => workflow.id !== workflowId);
        flowTeam.update({ workflows: teamWorkflows });
        return schema.db.summaries.remove({ id: workflowId });
      });

      // Workflow Revision
      this.get(serviceUrl.getWorkflowRevision({ workflowId: ":workflowId" }), (schema, request) => {
        let { workflowId } = request.params;
        if (workflowId) {
          return schema.revisions.findBy({ workFlowId: workflowId });
        } else {
          return {};
        }
      });

      this.get(
        serviceUrl.getWorkflowRevision({ workflowId: ":workflowId", revisionNumber: ":revisionNumber" }),
        (schema, request) => {
          let { workflowId, revisionNumber } = request.params;
          if (revisionNumber) {
            return schema.revisions.findBy({ workFlowId: workflowId, version: revisionNumber });
          } else {
            return {};
          }
        }
      );

      this.get(serviceUrl.getWorkflowTaskTemplates({ workflowId: ":workflowId" }), (schema, request) => {
        return schema.db.tasktemplate;
      });

      this.post(serviceUrl.postCreateWorkflowRevision({ workflowId: ":workflowId" }), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let { workflowId } = request.params;
        let revision = { ...body, workFlowId: workflowId };
        return schema.revisions.create(revision);
      });

      //Workflow Config Cron

      this.get(`${BASE_URL}/workflow/validate/cron`, () => {
        return {
          valid: true,
        };
      });

      // Workflow Properties
      this.patch(serviceUrl.patchUpdateWorkflowProperties({ workflowId: ":workflowId" }), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let { workflowId } = request.params;
        let summary = schema.summaries.find(workflowId);
        summary.update({ properties: body });
        return summary;
      });

      // Workflow Changelog
      this.get(serviceUrl.getWorkflowChangelog({ workflowId: ":workflowId" }), (schema, request) => {
        let { workflowId } = request.params;
        return schema.changelogs.where({ workflowId });
      });

      //Workflow Available Parameters
      this.post(serviceUrl.workflowAvailableParameters({ workflowId: ":workflowId" }), (schema, request) => {
        return schema.db.availableParameters[0].data;
      });
      /**
       * Activity
       */
      this.get(serviceUrl.getWorkflowRuns({ query: null }), (schema) => {
        return schema.db.workflowRuns[0];
      });

      this.get(serviceUrl.getWorkflowRunCount({ query: null }), (schema, request) => {
        return schema.db.workflowRunCount[0];
      });

      this.get(serviceUrl.getWorkflowExecution({ executionId: ":id" }), (schema, request) => {
        return schema.db.workflowExecution[0];
      });

      this.delete(serviceUrl.deleteCancelWorkflow({ executionId: ":id" }), (schema, request) => {
        return {};
      });

      /**
       * Actions
       */
      this.get(serviceUrl.getActionsSummary({ query: null }), (schema) => {
        return schema.db.actionsSummary[0];
      });

      this.get(serviceUrl.getActions({ query: null }), (schema, request) => {
        const { type } = request.queryParams;
        if (type === "approval") return schema.db.approvals[0];
        if (type === "task") return schema.db.manualTasks[0];
        return {};
      });

      this.put(serviceUrl.putWorkflowAction(), () => {
        return {};
      });

      /**
       * Approvers Group
       */
      this.get(serviceUrl.resourceApproverGroups({ teamId: ":teamId" }), (schema) => {
        return schema.db.approverGroups;
      });

      this.get(serviceUrl.getTeamMembers({ teamId: ":teamId" }), (schema) => {
        return schema.db.teamApproverUsers;
      });

      //Delete approver group
      this.delete(serviceUrl.resourceApproverGroups({ teamId: ":teamId", groupId: ":groupId" }), (schema, request) => {
        const { groupId } = request.params;
        const approverGroup = schema.approverGroups.find(groupId);
        approverGroup.destroy();
      });

      //Create approver group
      this.post(serviceUrl.resourceApproverGroups({ teamId: ":teamId" }), (schema, request) => {
        const body = JSON.parse(request.requestBody);
        schema.approverGroups.create({ groupId: uuid(), ...body });
        return schema.approverGroups.all();
      });

      //Update approver group
      this.put(serviceUrl.resourceApproverGroups({ teamId: ":teamId" }), (schema, request) => {
        return {};
      });

      /**
       * Manage Team
       */

      this.post(serviceUrl.postTeamValidateName(), (schema, request) => {
        return new Response(422, {}, { errors: [ 'Name is already taken'] });
      });

      this.get(serviceUrl.getTeam({ teamId: ":teamId" }), (schema, request) => {
        let { teamId } = request.params;
        return schema.manageTeamDetails.findBy({ id: teamId });
      });

      this.patch(serviceUrl.getTeam({ teamId: ":teamId" }), (schema, request) => {
        let { teamId } = request.params;
        let body = JSON.parse(request.requestBody);
        let activeTeam = schema.manageTeamDetails.findBy({ id: teamId });
        let activeUsers = activeTeam.users.filter((user) => body.includes(user.id));
        activeTeam.update({ users: activeUsers });
        return activeTeam;
      });

      this.put(serviceUrl.getTeam({ teamId: ":teamId" }), (schema, request) => {
        let { teamId } = request.params;
        let body = JSON.parse(request.requestBody);
        let summary = schema.manageTeamDetails.findBy({ id: teamId });
        summary.update(body);
        return summary;
      });

      this.patch(serviceUrl.getManageTeamLabels({ teamId: ":teamId" }), (schema, request) => {
        let { teamId } = request.params;
        let body = JSON.parse(request.requestBody);
        let activeTeam = schema.manageTeamDetails.findBy({ id: teamId });
        activeTeam.update({ labels: body });
        return activeTeam;
      });

      this.post(serviceUrl.getManageTeamsCreate(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        const teams = schema.teams.first();
        const updatedRecords = teams.records.concat({ id: uuid(), isActive: true, ...body });
        teams.update({ records: updatedRecords });
        return {};
      });

      /**
       * Manage Users
       */

      this.get(serviceUrl.getUsers({ query: null }), (schema, request) => {
        const { query } = request.queryParams;
        const userData = schema.db.users[0];
        if (query) {
          userData.content =
            userData.content.filter((user) => user.name.includes(query) || user.email.includes(query)) ?? [];
        }
        return userData;
      });

      this.get(serviceUrl.getUser({ userId: ":userId" }), (schema, request) => {
        const { userId } = request.params;
        const user = schema.db.users[0].content.find((user) => user.id === userId);
        return user;
      });

      this.patch(serviceUrl.getUser({ userId: ":userId" }), (schema, request) => {
        const { userId } = request.params;
        let body = JSON.parse(request.requestBody);
        const users = schema.users.first();
        const updatedRecords = [];
        for (let user of users.records) {
          if (user.id === userId) {
            user = user = { ...user, ...body };
          }
          updatedRecords.push(user);
        }
        users.update({ records: updatedRecords });
        return {};
      });

      /**
       * Quotas
       */

      this.patch(serviceUrl.getTeamQuotas({ id: ":id" }), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        const quotas = schema.quotas.first();

        const key = Object.keys(body)[0];
        const value = Object.values(body)[0];

        quotas.update({ [key]: value });
        return {};
      });

      this.put(serviceUrl.putTeamQuotasDefault({ id: ":id" }), (schema, request) => {
        const quotas = schema.quotas.first();
        const defaultConfig = {
          maxWorkflowCount: 20,
          maxWorkflowExecutionMonthly: 150,
          maxWorkflowStorage: 10,
          maxWorkflowExecutionTime: 30,
          maxConcurrentWorkflows: 4,
        };

        quotas.update(defaultConfig);
        return {};
      });

      this.put(serviceUrl.getTeamQuotas({ id: ":id" }), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        const quotas = schema.quotas.first();
        quotas.update(body);
        return {};
      });

      this.get(serviceUrl.getTeamQuotaDefaults(), (schema, request) => {
        return {
          maxWorkflowCount: 20,
          maxWorkflowExecutionMonthly: 150,
          maxWorkflowStorage: 10,
          maxWorkflowExecutionTime: 30,
          maxConcurrentWorkflows: 4,
        };
      });

      /**
       *  Manage Settings
       * */

      this.get(serviceUrl.resourceSettings(), (schema) => {
        return schema.settings.all();
      });

      this.put(serviceUrl.resourceSettings(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        const settings = schema.settings.all();
        settings.update(body[0]);
        return schema.settings.all();
      });

      /**
       * Manage and Administer Tokens
       */
      this.get(serviceUrl.getTeamTokens({ teamId: ":teamId" }), (schema) => {
        return schema.db.tokens;
      });

      this.get(serviceUrl.getGlobalTokens(), (schema) => {
        return schema.db.tokens;
      });

      this.delete(serviceUrl.deleteToken({ tokenId: ":tokenId" }), (schema, request) => {
        return {};
      });

      this.post(serviceUrl.postGlobalToken(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let newToken = {
          ...body,
          creatorId: "1",
          creationDate: Date.now(),
          creatorName: "Test User",
          tokenValue: "testglobal",
        };
        return schema.tokens.create(newToken);
      });

      this.post(serviceUrl.postTeamToken(), (schema, request) => {
        let body = JSON.parse(request.requestBody);
        let newToken = {
          ...body,
          creatorId: "1",
          creationDate: Date.now(),
          creatorName: "Test User",
          tokenValue: "testteam",
        };
        return schema.tokens.create(newToken);
      });

      /**
       * Workflow Templates
       */
      this.post(serviceUrl.postDuplicateWorkflow({ workflowId: ":workflowId" }), (schema, request) => {
        return {};
      });
      /**
       * TODO
       */
    },
  });
}
