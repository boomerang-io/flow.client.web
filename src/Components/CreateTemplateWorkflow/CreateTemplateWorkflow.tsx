//@ts-nocheck
import React from "react";
import { useFeature } from "flagged";
import { useMutation, useQueryClient, useQuery } from "react-query";
import { useHistory } from "react-router-dom";
import { Button } from "@carbon/react";
import { ModalFlow, notify, ToastNotification } from "@boomerang-io/carbon-addons-boomerang-react";
import CreateWorkflowContent from "./CreateWorkflowContent";
import CreateWorkflowTemplates from "./CreateWorkflowTemplates";
import { appLink } from "Config/appConfig";
import { serviceUrl, resolver } from "Config/servicesConfig";
import { Template, Add } from "@carbon/react/icons";
import { FlowTeam, ModalTriggerProps } from "Types";
import { FeatureFlag } from "Config/appConfig";
import styles from "./createWorkflow.module.scss";

interface CreateTemplateWorkflowProps {
  team: FlowTeam;
}

const CreateTemplateWorkflow: React.FC<CreateTemplateWorkflowProps> = ({ team }) => {
  const queryClient = useQueryClient();
  const history = useHistory();
  const workflowQuotasEnabled: boolean = useFeature(FeatureFlag.WorkflowQuotasEnabled);
  const workflowTemplatesUrl = serviceUrl.workflowTemplates();
  const getTaskTemplatesUrl = serviceUrl.getTaskTemplates({ query: "" });

  const {
    data: workflowTemplatesData,
    isLoading: workflowTemplatesIsLoading,
    error: workflowTemplatesError,
  } = useQuery({
    queryKey: workflowTemplatesUrl,
    queryFn: resolver.query(workflowTemplatesUrl),
  });

  const {
    data: taskTemplatesData,
    error: taskTemplatesError,
    isLoading: taskTempaltesAreLoading,
  } = useQuery({
    queryKey: getTaskTemplatesUrl,
    queryFn: resolver.query(getTaskTemplatesUrl),
  });

  const {
    mutateAsync: createTemplateWorkflowMutator,
    isLoading: createTemplateWorkflowIsLoading,
    error: createTemplateWorkflowError,
  } = useMutation(resolver.postTemplateWorkflow);

  const handleCreateWorkflow = async (selectedTemplateId: string, requestBody: any) => {
    try {
      const { data: newWorkflow } = await createTemplateWorkflowMutator({
        workflowId: selectedTemplateId,
        body: requestBody,
      });
      const workflowId = newWorkflow.id;

      history.push(appLink.editorDesigner({ workflowId }));
      notify(
        <ToastNotification
          kind="success"
          title="Create Workflow"
          subtitle="Successfully created workflow from template"
        />
      );
      queryClient.invalidateQueries(serviceUrl.getTeams());
      return;
    } catch (e) {
      console.log(e);
      return;
      //no-op
    }
  };
  const isLoading = createTemplateWorkflowIsLoading;

  return (
    <ModalFlow
      composedModalProps={{ containerClassName: styles.modalContainer }}
      modalTrigger={({ openModal }: ModalTriggerProps) => (
        <Button
          onClick={openModal}
          renderIcon={Add}
          kind="ghost"
          size="md"
          data-testid="workflows-create-workflow-template-button"
        >
          <Template className={styles.addIcon} />
          <p className={styles.text}>Templates</p>
        </Button>
      )}
      modalHeaderProps={{
        title: "Create Workflow from Template",
        subtitle: "Get started by leveraging this template",
      }}
    >
      <CreateWorkflowTemplates
        templatesError={workflowTemplatesError || taskTemplatesError}
        isLoading={workflowTemplatesIsLoading || taskTempaltesAreLoading}
        workflowTemplates={workflowTemplatesData}
        taskTemplates={taskTemplatesData}
      />
      <CreateWorkflowContent
        createWorkflow={handleCreateWorkflow}
        createError={createTemplateWorkflowError}
        isLoading={isLoading}
        team={team}
        workflowQuotasEnabled={workflowQuotasEnabled}
      />
    </ModalFlow>
  );
};

export default CreateTemplateWorkflow;
