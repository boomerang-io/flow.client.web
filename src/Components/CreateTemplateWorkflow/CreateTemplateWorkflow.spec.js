/* eslint-disable jest/no-commented-out-tests */
import React from "react";
import { Route } from "react-router-dom";
import CreateTemplateWorkflow from ".";
import { queryCaches } from "react-query";
import { AppPath, appLink } from "Config/appConfig";
import { startApiServer } from "ApiServer";

const team = {
  id: "1234",
  name: "Lucas' team",
  workflows: [
    {
      id: "hQDkX9v",
      name: "lucas-workflow-1",
      description: "blablabla",
      status: "published",
      icon: "utility",
    },
    {
      id: "456",
      name: "lucas-workflow-2",
      description: "blablabla",
      status: "published",
      icon: "utility",
    },
    {
      id: "789",
      name: "lucas-workflow-3",
      description: "blablabla",
      status: "draft",
      icon: "secure",
    },
  ],
};

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
  queryCaches.forEach((queryCache) => queryCache.clear());
});

const props = {
  scope: "user",
  teams: [team],
};

describe("CreateTemplateWorkflow --- Snapshot Test", () => {
  test("Capturing Snapshot of CreateTemplateWorkflow", () => {
    const { baseElement } = rtlContextRouterRender(
      <Route path={AppPath.WorkflowsTeams}>
        <CreateTemplateWorkflow {...props} />
      </Route>,
      { route: appLink.workflowsTeams() }
    );
    expect(baseElement).toMatchSnapshot();
  });
});

//Need to look a bit more on the issue here
// describe("CreateTemplateWorkflow --- RTL", () => {
//   it("Create Template", async () => {
//     const { getByText, findByTestId } = rtlContextRouterRender(
//       <Route path={AppPath.WorkflowsTeams}>
//         <CreateTemplateWorkflow {...props} />
//       </Route>,
//       { route: appLink.workflowsTeams() }
//     );
//     fireEvent.click(getByText(/Templates/));
//     await findByTestId("initial-template-screen");
//     expect(await findByText("Next")).toBeDisabled();
//     fireEvent.click(getByText(/Test name/));
//     expect(getByText("Next")).toBeEnabled();
//     fireEvent.click(getByText(/Next/));
//     fireEvent.click(getByText(/^Create$/i));
//     expect(await findByText("Successfully created workflow from template")).toBeInTheDocument();
//   });
// });
