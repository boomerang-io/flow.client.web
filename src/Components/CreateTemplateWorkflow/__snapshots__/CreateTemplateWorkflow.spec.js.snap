// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTemplateWorkflow --- Snapshot Test Capturing Snapshot of CreateTemplateWorkflow 1`] = `
<body
  id="app"
>
  <div>
    <button
      class="bx--btn bx--btn--field bx--btn--ghost"
      data-testid="workflows-create-workflow-template-button"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="addIcon"
        fill="currentColor"
        focusable="false"
        height="32"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M26 6v4H6V6H26m0-2H6A2 2 0 004 6v4a2 2 0 002 2H26a2 2 0 002-2V6a2 2 0 00-2-2zM10 16V26H6V16h4m0-2H6a2 2 0 00-2 2V26a2 2 0 002 2h4a2 2 0 002-2V16a2 2 0 00-2-2zM26 16V26H16V16H26m0-2H16a2 2 0 00-2 2V26a2 2 0 002 2H26a2 2 0 002-2V16a2 2 0 00-2-2z"
        />
      </svg>
      <p
        class="text"
      >
        Templates
      </p>
      <svg
        aria-hidden="true"
        class="bx--btn__icon"
        fill="currentColor"
        focusable="false"
        height="32"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
        />
      </svg>
    </button>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
